client/service/src/client/block_rules.rs

```diff
@@ -30,18 +30,31 @@ use sc_client_api::{ForkBlocks, BadBlocks};
 pub enum LookupResult<B: BlockT> {
 	/// Specification rules do not contain any special rules about this block
 	NotSpecial,
-	/// The bock is known to be bad and should not be imported
+	/// The block is known to be bad and should not be imported
 	KnownBad,
+	/// The block is known not to be finalized
+	KnownUnfinalized,
```

Comment:
I'd call it something like `FinalityBlacklist`. "Unfinalized" sounds like it might still be finalized. Also "block that should not be finalized" practically means it should not be imported as well, right @rphmeier ? So from the perspective of `BlockRules` it is the same as `KnownBad`

Reply:
Yeah, it should not be imported.

Reply:
Changed this to directly insert it into BadBlock so it won't get imported.